sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinHelper.scala

```diff
@@ -218,11 +222,27 @@ object StreamingSymmetricHashJoinHelper extends Logging {
           attributesToFindStateWatermarkFor = AttributeSet(oneSideInputAttributes),
           attributesWithEventWatermark = AttributeSet(otherSideInputAttributes),
           condition,
-          eventTimeWatermarkForEviction)
-        val inputAttributeWithWatermark = oneSideInputAttributes.find(_.metadata.contains(delayKey))
-        val expr = watermarkExpression(inputAttributeWithWatermark, stateValueWatermark)
-        expr.map(JoinStateValueWatermarkPredicate.apply _)
+          eventTimeWatermarkForEviction
+        )
+
+        val attributesInCondition = AttributeSet(
+          condition.get.collect { case a: AttributeReference => a }
```

Comment:
add a comment. What are we doing here? I don't know why it is the right thing to do.

Reply:
Is every `Attribute` an `AttributeReference`? (I don't know much about this class hierarchy)

Reply:
Comment added.

No, the hierarchy is:

```
Attribute --> AttributeReference --> ColumnarAttributeReference
|
| --> PrettyAttribute (used only for debugging)
```

So it's safe to just use an `AttributeReference` in the collect. We also do this when we generate the state value watermark predicate: https://github.com/apache/spark/blob/f3a6ca9e2c476fe418679f82b0c5148d69d0fbda/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/StreamingJoinHelper.scala#L139